roles/check_changelog_fragment/tasks/main.yml

```diff
@@ -0,0 +1,73 @@
+---
+# RULES
+# * a pull-request needs a new changelog fragment
+# * unless:
+#   * it is a "New plugin" pull-request
+#   * unless:
+#     * the PR modifies existing file(s)
+#     * the PR deletes existing file(s)
+#     * the PR renames existing file(s)
+
+
+# Rebasing ensures diffs are related to the PR, and only to the PR. If the task
+# fails, this is because of conflicts, i.e. the PR is not ready for merging (or
+# one of git_directory or target_branch is incorrect).
+- name: Rebase the PR branch on the top of the target branch
+  command:
+    chdir: "{{ check_changelog_fragment__git_directory }}"
+    cmd: "git rebase {{ check_changelog_fragment__target_branch }}"
+  register: check_changelog_fragment__rebase
+  changed_when: check_changelog_fragment__rebase.stdout_lines | length > 1
+
+
+- name: Look for any diff between current and target branches
```

Comment:
there should always be a diff right? As the PR running will have a new commit?

Reply:
yes, there should be a diff, except in the use case explained above by @goneri (playing the role in scheduled jobs against the default branch itself, so with an empty diff)